// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/AzureAccount.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AzureAccount = void 0;
/**
 * Datadog-Azure integrations configured for your organization.
 */
class AzureAccount {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AzureAccount.attributeTypeMap;
    }
}
exports.AzureAccount = AzureAccount;
/**
 * @ignore
 */
AzureAccount.attributeTypeMap = {
    appServicePlanFilters: {
        baseName: "app_service_plan_filters",
        type: "string",
    },
    automute: {
        baseName: "automute",
        type: "boolean",
    },
    clientId: {
        baseName: "client_id",
        type: "string",
    },
    clientSecret: {
        baseName: "client_secret",
        type: "string",
    },
    containerAppFilters: {
        baseName: "container_app_filters",
        type: "string",
    },
    cspmEnabled: {
        baseName: "cspm_enabled",
        type: "boolean",
    },
    customMetricsEnabled: {
        baseName: "custom_metrics_enabled",
        type: "boolean",
    },
    errors: {
        baseName: "errors",
        type: "Array<string>",
    },
    hostFilters: {
        baseName: "host_filters",
        type: "string",
    },
    newClientId: {
        baseName: "new_client_id",
        type: "string",
    },
    newTenantName: {
        baseName: "new_tenant_name",
        type: "string",
    },
    tenantName: {
        baseName: "tenant_name",
        type: "string",
    },
};
